var/spack/repos/builtin/packages/hdf-eos2/package.py

```diff
@@ -54,6 +55,11 @@ class HdfEos2(AutotoolsPackage):
 
     # Build dependencies
     depends_on("hdf")
+    # Because hdf always depends on zlib and jpeg in spack, the tests below in configure_args
+    # (if "jpeg" in self.spec:) always returns true and hdf-eos2 wants zlib and jpeg, too.
+    depends_on("zlib")
```

Comment:
Should this be `zlib-api` ? The test down below (`if "zlib" in self.spec`) would probably not trigger on `"zlib-api"`, because of the way the dependency is encoded in `hdf`. (The tests in lines 112-117 trigger on the specs of `hdf` in the absence of the newly introduced lines 58-62.)

Reply:
I think it should

Reply:
I checked https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/hdf/package.py and it also says
```
depends_on("zlib-api")
```
so I will go ahead and make the change here.

Reply:
Done.